### PR TITLE
Do not expose the expected code when it does not match

### PR DESCRIPTION
**Pull request by julien-nc**
Only expose the expected code when the instance is in debug mode.

### CODE REVIEW

**Review by juliushaertl — APPROVED** [review]

File: lib/Controller/Id4meController.php [hunk 1]
```diff
@@ -173,11 +179,16 @@ public function code($state = '', $code = '', $scope = '') {
...
 		if ($this->session->get(self::STATE) !== $state) {
+			$responseData = [
+				'error' => 'invalid_state',
+				'error_description' => 'received state does not match the expected value',
+			];
+			if ($this->config->getSystemValueBool('debug', false)) {
+				$responseData['got'] = $state;
+				$responseData['expected'] = $this->session->get(self::STATE);
+			}
 			// TODO show page with forbidden
```
Comment (nickvergessen) [inline lib/Controller/Id4meController.php:190]: it's returning a forbidden, so we can remove this? It shows an error in the frontend?

Comment (julien-nc) [reply lib/Controller/Id4meController.php:190]: I guess the comment means "returning a proper forbidden page with a nice template".
Now it only shows the raw returned json.

Comment (juliushaertl) [reply lib/Controller/Id4meController.php:190]: Maybe we can move to an error template page directly as seeing json in the browser is far from nice ;)

Comment (julien-nc) [reply lib/Controller/Id4meController.php]: :+1: Done

**Pull request merged**